### PR TITLE
feat: use DTLS for relay

### PR DESCRIPTION
**Pull request by unity-ptrottier**
### Description (*)
Use DTLS when using relay instead of raw UDP. The peer couldn't see the heroes on my end with or without DTLS, but @j-bro didn't see the problem.

### Issue Number(s) (*)

Fixes issue(s):
https://jira.unity3d.com/browse/MPR-1178

### CODE REVIEW

**Review by simon-lemay-unity — DISMISSED** [review]

**Comment by simon-lemay-unity** [comment]
Might be a good occasion to convert the `SetRelayServerData` calls to `SetHostRelayData` and `SetClientRelayData` (the signatures are the same, just `SetHostRelayData` doesn't need to pass a null value for the host connection data). Otherwise looks good to me.

**Comment by unity-ptrottier** [comment]
> Might be a good occasion to convert the `SetRelayServerData` calls to `SetHostRelayData` and `SetClientRelayData` (the signatures are the same, just `SetHostRelayData` doesn't need to pass a null value for the host connection data). Otherwise looks good to me.

Will do!

**Review by simon-lemay-unity — DISMISSED** [review]

**Comment by SamuelBellomo** [comment]
putting on hold until we have our GDC branch created. We don't want newer features on a branch we're trying to stabilize. If we want to playtest DTLS in particular, let's create a build from this branch instead from develop

**Comment by SamuelBellomo** [comment]
To clarify, we'll be able to merge this, but only after GDC branch is created

**Comment by unity-ptrottier** [comment]
> To clarify, we'll be able to merge this, but only after GDC branch is created

Thanks Samuel, do you know when this branching will be done?

**Review by j-bro — DISMISSED** [review]

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs
Comment (bgagnon) [inline Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs:43]: 🚫 instead of finding `e.Secure`, compare against `"dtls"`. The `e.Secure` might be enabled when/if Relay supports TLS (secured TCP), and this code would break.

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs
Comment (bgagnon) [inline Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs:67]: Check for `"dtls"` here as well.

Comment (unity-ptrottier) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs]: [680588e](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/485/commits/680588e09f045f797ed8706a19d8e6bf1038dda1)

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs
Comment (unity-ptrottier) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs]: [680588e](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/485/commits/680588e09f045f797ed8706a19d8e6bf1038dda1)

**Review by bgagnon — APPROVED** [review]
🚀

File: Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs:241]: for clarity in a sample like this, I usually like to have named parameters for simple types, so readers know what that "true" is useful for
```suggestion
                        utp.SetClientRelayData(ipv4Address, port, allocationIdBytes, key, connectionData, hostConnectionData, isSecure: true);
```

File: Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/GameNetPortal.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/GameNetPortal.cs:232]:
```suggestion
                        utp.SetHostRelayData(ipv4Address, port, allocationIdBytes, key, connectionData, isSecure: true);
```

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs
Comment (unity-ptrottier) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/Net/UnityRelayUtilities.cs]: e64afd61

**Review by SamuelBellomo — APPROVED** [review]

File: Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs
Comment (bgagnon) [reply Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs]: For the same reason [mentioned here](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/485#discussion_r812967013), I think UTP won't really have a choice but to adopt the string-based (or enum) connection types instead of relying on a boolean.

Comment (simon-lemay-unity) [reply Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs]: Good point, although unfortunately I'm not sure we can/want to break the API at this point. There's also no plan to support TCP in UTP (except through WebSockets eventually), so for now I think we can live with the limitation.

**Comment by SamuelBellomo** [comment]
GDC branch created, once conflicts are solved, we can merge this :)

**Comment by SamuelBellomo** [comment]
Let's make sure this isn't broken after Lobby integration

**Review by j-bro — APPROVED** [review]
🚀

**Pull request merged**
